students/SLammertink/Lesson04/mailroom_part2.py

```diff
@@ -0,0 +1,101 @@
+#! /usr/bin/env python3
+# Author: SLammertink
+# Lesson 4 mailroom part 2 exercise
+
+'''
+Update your mailroom program to:
+
+Use dicts where appropriate
+See if you can use a dict to switch between the users selections. See Using a Dictionary to switch for what this means.
+Try to use a dict and the .format() method to do the letter as one big template rather than building up a big string in parts.
+'''
+
+import collections
+import pathlib
+
+donors = collections.defaultdict(list)
+
+# Create the switch dictionary:
+switch_dict = { "1" : "Sending a Thank You" , "2" : "Creating a Report", "3" : "Send letters to everyone", "4" : "Quit"}
+donors = {'Sukhmani Travers': [250 ,125, 1000], 'Sebastien Mayo': [500, 35], 'Aryan Davila' : [25, 45, 55, 120], 'Zayan Langley' : [ 1000, 10, 250], 'Charlotte Bates' : [25, 10, 5, 35, 75, 100] }
+
+def pretty():
+    ''' a function that adds lines and spaces, just for the looks '''
+    print('')
+    print('{:-<43}'.format(''))
+    print('')
+
+def switch():
+    ''' a function that ask the user to input a number that
+    corresponds to an action that will be executed '''
+    pretty()
+    while True:
+        print("Please choose from the following menu: \n")
+        for k, v in switch_dict.items():
+            print(k, v)
+        pretty()
+        choice = input("Please enter the choice you want to make: ")
+        pretty()
+        if choice == "list":
```

Comment:
you could add this option to your `switches` dictionary to avoid having a special case in your main loop.